exercises/practice/minima/minima_test.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+from minima import minima
+import z3
+
+def toFloat(x: z3.RatNumRef) -> float:
+    """"""
+    if(isinstance(x, float) or isinstance(x, int)):
+        return x
+    x = x.as_fraction()
+    return float(x.numerator) / float(x.denominator)
+
+def createPoly(a: float, b: float, c: float, d: float) -> "Callable":
+    """Returns a function representing a polynomial that can be evaluated for an x."""
+    return lambda x: a * x ** 3 + b * x ** 2 + c * x + d
+
+class MinimaTest(unittest.TestCase):
+    def checkIsMin(self, f: "Callable", x: float) -> None:
+        """Checks with unit test asserts that the given x value is a minimum."""
+        self.assertLess(f(x), f(x - 0.001))
+        self.assertLess(f(x), f(x + 0.001))
```

Comment:
How deep into edge cases are you going to get?  What about (0, 0, 0, 5)?  If not, maybe "nonzero" parameters should be specified in the instructions.

Reply:
I think it's just best to say in the instructions that the function will have a minima somewhere. Basically non-zero, but that could be ambiguous if it's one or all.